lib/loadGltfUris.js

```diff
@@ -115,9 +115,10 @@ function isTransparent(image) {
 function readImageFromFile(object, name, uriPath) {
     return fsReadFile(uriPath)
         .then(function(data) {
+            var ext = path.extname(uriPath);
             object.extras._pipeline.source = data;
-            object.extras._pipeline.extension = path.extname(uriPath);
-            if (name === 'images') {
+            object.extras._pipeline.extension = ext;
+            if (name === 'images' && ext !== '.ktx' && ext !== '.gif') {
```

Comment:
Should this check stay in `generateJimpImage`? If the image is embedded it will not reach this code.